### PR TITLE
Add OpenStack Designate dns provider

### PR DESCRIPTION
**Pull request by databus23**
This PR adds support for OpenStack Designate as a dns provider.

Manually tested and verified to work on Converged Cloud EE.

Let me know what you think.

/cc @vasu1124 @BugRoger

### CODE REVIEW

File: charts/gardener/templates/secret-default-domain.yaml [hunk 1]
```diff
@@ -18,7 +18,16 @@ metadata:
     dns.garden.sapcloud.io/hostedZoneID: {{ ( required ".controller.defaultDomains[].hostedZoneID is required" $domain.hostedZoneID ) }}
 type: Opaque
 data:
+{{- if eq $domain.provider "aws-route53" }}
```
Comment (rfranzke) [inline charts/gardener/templates/secret-default-domain.yaml:21]: Could you please also give a hint that `openstack-designate` is supported and which keys are required [here](https://github.com/gardener/gardener/blob/master/charts/gardener/values.yaml#L76) (similar to `google-clouddns`)? Can you please also add an example manifest [here](https://github.com/gardener/gardener/blob/master/example/10-secret-default-domain.yaml)?

File: charts/gardener/templates/secret-internal-domain.yaml [hunk 2]
```diff
@@ -16,6 +16,15 @@ metadata:
     dns.garden.sapcloud.io/hostedZoneID: {{ ( required ".controller.internalDomain.hostedZoneID is required" .Values.controller.internalDomain.hostedZoneID ) }}
 type: Opaque
 data:
+{{- if eq .Values.controller.internalDomain.provider "aws-route53" }}
```
Comment (rfranzke) [inline charts/gardener/templates/secret-internal-domain.yaml:19]: Could you please also give a hint that `openstack-designate` is supported and which keys are required [here](https://github.com/gardener/gardener/blob/master/charts/gardener/values.yaml#L65) (similar to `google-clouddns`)? Can you please also add an example manifest [here](https://github.com/gardener/gardener/blob/master/example/10-secret-internal-domain.yaml)?

File: plugin/pkg/shoot/dnshostedzone/admission.go [hunk 3]
```diff
@@ -174,6 +175,17 @@ func verifyHostedZoneID(shoot *garden.Shoot, secretBindingLister gardenlisters.S
 		if !serviceAccountJSONFound {
 			return fmt.Errorf("specifying the `.spec.dns.hostedZoneID` field is only possible if the cloud provider secret or the secret referenced in .spec.dns.secretName contains credentials for Google CloudDNS (%s)", gcpbotanist.ServiceAccountJSON)
 		}
+	case garden.DNSOpenstackDesignate:
```
Comment (rfranzke) [inline plugin/pkg/shoot/dnshostedzone/admission.go:178]: Could you please add a test for the OpenStack case (similar to [that](https://github.com/gardener/gardener/blob/master/plugin/pkg/shoot/dnshostedzone/admission_test.go#L210) and [that](https://github.com/gardener/gardener/blob/master/plugin/pkg/shoot/dnshostedzone/admission_test.go#L175))?

**Review by rfranzke — CHANGES_REQUESTED** [review]
Thank you a lot for your efforts @databus23 and @BugRoger, it looks very good!
Could you please add the mentioned example configuration/manifests and a small test in the admission controller before we get this merged?

**Comment by databus23** [comment]
@rfranzke I updated the PR according to your feedback.

**Review by rfranzke — APPROVED** [review]
LGTM, thank you!

**Pull request merged**